src/app/beer_garden/db/mongo/pruner.py

```diff
@@ -6,57 +6,208 @@
 from brewtils.errors import ModelValidationError
 from brewtils.models import Event, Events
 from brewtils.schema_parser import SchemaParser
-from brewtils.stoppable_thread import StoppableThread
 from mongoengine import Q
 from mongoengine.errors import DoesNotExist
 
+import beer_garden.config as config
 from beer_garden.db.mongo.models import File, RawFile, Request
 from beer_garden.db.mongo.parser import MongoParser
 from beer_garden.events import publish
 
+logger = logging.getLogger(__name__)
+
+display_name = "Mongo Pruner"
+
+
+def run_pruner(tasks, ttl_name):
+    current_time = datetime.utcnow()
+
+    if tasks:
+        for task in tasks:
+            delete_older_than = current_time - task["delete_after"]
+
+            query = Q(**{task["field"] + "__lt": delete_older_than})
+            if task.get("additional_query", None):
+                query = query & task["additional_query"]
+
+            logger.debug(
+                "Removing %s %ss older than %s"
+                % (ttl_name, task["collection"].__name__, str(delete_older_than))
+            )
+
+            if task["batch_size"] > 0:
+                while (
+                    task["batch_size"]
+                    < task["collection"].objects(query).no_cache().count()
+                ):
+                    logger.debug(
+                        "Removing %s from %ss older than %s, batched by %s"
+                        % (
+                            ttl_name,
+                            task["collection"].__name__,
+                            str(delete_older_than),
+                            str(task["batch_size"]),
+                        )
+                    )
+                    task["collection"].objects(query).limit(
+                        task["batch_size"]
+                    ).no_cache().delete()
+
+            num = task["collection"].objects(query).no_cache().delete()
+            if num:
+                logger.debug(
+                    "Deleted %s %s from %ss"
+                    % (num, ttl_name, task["collection"].__name__)
```

Comment:
This only works for files. Delete embedded document in mongoengine is the only delete method that returns a count.